File: ems-backend/src/main/java/net/javaguides/emsbackend/service/EmployeeService.java
package net.javaguides.emsbackend.service;

import java.util.List;

import net.javaguides.emsbackend.dto.EmployeeDto;


public interface EmployeeService {
	EmployeeDto createEmployee(EmployeeDto employeeDto);
	
	//get all users
	List<EmployeeDto> getEmployees();
	
	//get employee by id
	EmployeeDto getEmployee(Long id);
	
	//update employee
	EmployeeDto updateEmployee(Long id, EmployeeDto employeeDto);
	
	//Delete Employee By Id
	void deleteEmployee(Long id);
}

File: frontend/src/services/EmployeeService.js
import axios from "axios"

const RESR_API_BASE_URL = 'http://localhost:8080/api/employees';

export const listEmployees = () =>  axios.get(RESR_API_BASE_URL);

export const createEmployee = (employee) => axios.post(RESR_API_BASE_URL, employee);

export const getEmployee = (id) => axios.get(RESR_API_BASE_URL + '/' + id);

export const updateEmployee = (id, employee) => axios.put(RESR_API_BASE_URL + "/" + id, employee);

export const deleteEmployee = (id) => axios.delete(RESR_API_BASE_URL + '/' + id);